### PR TITLE
finish udp chat

### PR DESCRIPTION
**Pull request by intade**
task 3

### CODE REVIEW

File: Lab1/UdpChat/UdpChat.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        public static void Main(string[] args)
+        {
+
+            try
+            {
+                Console.WriteLine("Enter your username : ");
+                _name = Console.ReadLine();
+
+                Console.WriteLine("Enter the sending port : ");
+                _remotePort = int.Parse(Console.ReadLine() ?? throw new Exception("are you a tupiza?"));
```
Comment (gmltA) [inline Lab1/UdpChat/UdpChat.cs:28]: 🤣

File: Lab1/UdpChat/UdpChat.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                
+                Console.WriteLine("Enter the receiving port : ");
+                _localPort = int.Parse(Console.ReadLine() ?? throw new Exception("are you a tupiza?"));
+                
+                Console.WriteLine("Welcome to the club, body");
```
Comment (gmltA) [inline Lab1/UdpChat/UdpChat.cs:33]:
```suggestion
                Console.WriteLine("Welcome to the club, buddy");
```

**Review by gmltA — COMMENTED** [review]
Very good. But I don't see how you detect package loss.

**Comment by intade** [comment]
I send a hash so a user on the other side will be able to calculate the hash of a received messege and compare these two hashes

**Comment by intade** [comment]
Oh..I got what you are talking about... I missed it. I'm going to fix that

**Comment by intade** [comment]
i'have added a thread which handle messages queue. It means that after writing a message, one will be added to the queue. And the thread will be resending first message in the queue delayed until it receives a DataArrived status-message

**Review by gmltA — APPROVED** [review]

**Pull request merged**